File: mike/cpp/squareroot.cpp
#include <cmath>
#include <iostream>

using namespace std;

#define EPS 1e-7

int main() {
    double C, bottom, mid, top, ans;
    cin >> C;

    mid = top = sqrt(C);
    bottom = sqrt(C / 2);

    while (bottom < top) {
        mid = (bottom + top) / 2.0;
        ans = pow(mid, 2) + sqrt(mid);
        if (abs(ans - C) < EPS) break;
        (ans < C) ? bottom = mid : top = mid;
    }
    cout << mid << endl;
}

File: mike/python/squareroot.py
ε = 1e-7
C = float(input())

bottom = (C/2)**0.5
mid = top = C**0.5

while bottom < top:
	mid = (bottom + top) / 2
	equation = mid**2 + mid**0.5
	if abs(equation - C) < ε:
		break
	if equation < C:
		bottom = mid
	else:
		top = mid
	

print(round(mid, 6))
